src/crdt/internal/crdt-collection.ts

```diff
@@ -114,7 +114,7 @@ export class CRDTCollection<T extends Referenceable> implements CollectionModel<
     const operations = simplifyFastForwardOp(fastForwardOp) || [fastForwardOp];
 
     this.model.values = merged;
-    this.model.version = newClock;
+    this.model.version = newVersionMap;
```

Comment:
definitely outside the scope of this PR, just thinking out loud here. should we maybe also rename the `model.version` to `model.versionMap` for consistency?